### ISSUE TITLE
Hot-reloading not working

### ISSUE THREAD

**Issue by cdfa** [dmjio/miso#642]
**Describe the bug**
Hot-reloading with JSaddle is not working for me.

**To Reproduce**
1. `cd sample-app-jsaddle && nix-shell --run reload` 
2. `open http://localhost:8080`
3. Change something in Main.hs (for example text on button)
4. Change is not reflected on webpage and counter does not function any more

**Expected behavior**
Page is updated and counter still functions

**Logs**
```
All good (1 module, at 14:20:27)
Running test...

Hello World

Reloading...
  /home/cdfa/Projects/miso/sample-app-jsaddle/Main.hs

Intghc: ConnectionClosed
errupted.

All good (1 module, at 14:20:37)
Running test...
```

**Desktop (please complete the following information):**
 - OS: 5.4.89-1-MANJARO GNU/Linux
 - Browser: Firefox
 - Version: 85.0

**Additional info**
Everything works as expected after a page refresh.

**Comment by valyagolev** [comment]
same. I saw it working on Windows I think (WSL), but not on Mac OS. js console gets:

```
jsaddle.js:124 WebSocket is already in CLOSING or CLOSED state.
func @ jsaddle.js:124
eval @ VM62:32
```

**Comment by cdfa** [comment]
I've learned a bit more about the ghcjs ecosystem since posting this issue and I think doing a page refresh is inevitable.
I did find that using `runApp f = JSaddle.debug 3708 f` instead of `runApp f = Warp.runSettings ...` automatically refreshes for you.
I also came across a package [preserving state through reload](https://github.com/parsonsmatt/servant-persistent/pull/33) using `foreign-store`, but I haven't had time to look into it yet.
